src/Lykke.Service.KycReports.Services/Reports/KycReportingService.cs

```diff
@@ -0,0 +1,766 @@
+                            })
+                            .ToList();
+                        reportRows.AddRange(toResubmit);
+
+                        if (onBoarded.Count == 0 && declined.Count == 0 && toResubmit.Count == 0)
+                            await _reportRepository.InsertRow(KycOfficersPerformanceRow.MakeNoDataRow(startOfDay));
+                    }
+                }
+                catch (Exception ex)
+                {
+                    await _log.WriteErrorAsync("SrvReports", "GenerateKycOfficersPerformance (process day)", startOfDay.ToString(), ex);
+                    return false;
+                }
+
+
+            }
+
+            var clientIds = reportRows.Select(row => row.ClientId).Where(id => !string.IsNullOrEmpty(id)).Distinct();
+            var personalData = (await _personalDataService.GetAsync(clientIds)).ToDictionary(key => key.Id, val => val.Email);
+            foreach (var reportRow in reportRows)
+            {
+                try
+                {
+                    reportRow.ClientEmail = string.Empty;
+
+                    if (string.IsNullOrWhiteSpace(reportRow.ClientId))
+                        continue;
+
+                    if (personalData.ContainsKey(reportRow.ClientId))
+                        reportRow.ClientEmail = personalData[reportRow.ClientId];
+
+                    await _reportRepository.InsertRow(reportRow);
+                }
+                catch (Exception ex)
+                {
+                    await _log.WriteErrorAsync("SrvReports", "GenerateKycOfficersPerformance (insert new rows)", null, ex);
+                    return false;
+                }
+            }
+            
+            return true;
+        }
+
+        private async Task<List<KycStatusLogRecord>> GetKycStatusLogRecords(DateTime startDate, DateTime endDate)
+        {
+            ReportFilter f = new ReportFilter();
+            f.DatePeriod = new DatePeriod();
+            f.DatePeriod.StartDate = startDate;
+            f.DatePeriod.EndDate = endDate;
+
+            IEnumerable<IKycStatuschangeItem> items = await _kycStatusService.GetRecordsForPeriodAsync(f);
+            IEnumerable<Partner> partners = await _partnersService.GetPartnersAsync();
+            Dictionary<string, string> partnersDict = new Dictionary<string, string>();
+            foreach (Partner p in partners)
+            {
+                partnersDict[p.PublicId] = p.Name;
+            }
+
+            var auditLogEntities = (items)?
+                            .Select(item =>
+                            {
+                                var status = (KycStatus)item.CurrentStatus;
+                                var previousStatus = (KycStatus)item.PreviousStatus;
+                                
+                                const string boChanger = "BackOffice: ";
```

Comment:
All const strings should be moved to class private fields

Reply:
done